app/scripts/modules/core/src/pipeline/config/stages/evaluateVariables/EvaluateVariablesExecutionDetails.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+
+import { StageExecutionLogs, StageFailureMessage } from 'core/pipeline/details';
+import { ExecutionDetailsSection, IExecutionDetailsSectionProps } from 'core/pipeline/config/stages/core';
+
+export interface IEvaluatedVariables {
+  key: string;
+  value: string;
+}
+
+export function EvaluateVariablesExecutionDetails(props: IExecutionDetailsSectionProps) {
+  const {
+    stage: { context = {}, outputs = {} },
+  } = props;
+
+  const evaluatedVariables = context.variables ? (
+    <div>
+      <dl>
+        {context.variables.map((evalPair: IEvaluatedVariables) => (
+          <React.Fragment key={evalPair.key}>
```

Comment:
Is this necessary syntax because of the `key`?

Reply:
Yes. `< key={x.key}>` is not valid syntax.

Reply:
For context, this will allow for other fields like 'type' in the future.